src/main/java/twilightforest/entity/EntityTFBlockGoblin.java

```diff
@@ -24,22 +26,31 @@
 import net.minecraft.util.EnumHand;
 import net.minecraft.util.ResourceLocation;
 import net.minecraft.util.SoundEvent;
+import net.minecraft.util.math.MathHelper;
 import net.minecraft.util.math.Vec3d;
 import net.minecraft.world.World;
 import twilightforest.TFSounds;
 import twilightforest.TwilightForestMod;
 
 import java.util.List;
+import java.util.UUID;
 
 public class EntityTFBlockGoblin extends EntityMob implements IEntityMultiPart {
+	private static final UUID MODIFIER_UUID = UUID.fromString("5CD17E52-A79A-43D3-A529-90FDE04B181E");
+	private static final AttributeModifier MODIFIER = (new AttributeModifier(MODIFIER_UUID, "Drinking speed penalty", -0.25D, 0)).setSaved(false);
+
 
 	public static final ResourceLocation LOOT_TABLE = TwilightForestMod.prefix("entities/block_goblin");
 	private static final float CHAIN_SPEED = 16F;
 	private static final DataParameter<Byte> DATA_CHAINLENGTH = EntityDataManager.createKey(EntityTFBlockGoblin.class, DataSerializers.BYTE);
 	private static final DataParameter<Byte> DATA_CHAINPOS = EntityDataManager.createKey(EntityTFBlockGoblin.class, DataSerializers.BYTE);
+	private static final DataParameter<Boolean> ISTHROW = EntityDataManager.createKey(EntityTFBlockGoblin.class, DataSerializers.BOOLEAN);
 
 	private int recoilCounter;
 	private float chainAngle;
+	private float chainMoveLengh0;
```

Comment:
Is `chainMoveLengh0` required? It looks like it's always written to and never read from. Looks redundant I think.

Also `Lengh` -> `Length`. Missing the `t`